lib/ansible/modules/network/panos/panos_security_policy.py

```diff
@@ -0,0 +1,518 @@
+    description = module.params['description']
+    tag = module.params['tag']
+    from_zone = module.params['from_zone']
+    to_zone = module.params['to_zone']
+    source = module.params['source']
+    source_user = module.params['source_user']
+    destination = module.params['destination']
+    category = module.params['category']
+    application = module.params['application']
+    service = module.params['service']
+    hip_profiles = module.params['hip_profiles']
+    log_start = module.params['log_start']
+    log_end = module.params['log_end']
+    rule_type = module.params['rule_type']
+    action = module.params['action']
+
+    group_profile = module.params['group_profile']
+    antivirus = module.params['antivirus']
+    vulnerability = module.params['vulnerability']
+    spyware = module.params['spyware']
+    url_filtering = module.params['url_filtering']
+    file_blocking = module.params['file_blocking']
+    data_filtering = module.params['data_filtering']
+    wildfire_analysis = module.params['wildfire_analysis']
+
+    devicegroup = module.params['devicegroup']
+
+    commit = module.params['commit']
+
+    if devicegroup:
+        device = pandevice.panorama.Panorama(ip_address, username, password, api_key=api_key)
+        dev_grps = device.refresh_devices()
+
+        for grp in dev_grps:
+            if grp.name == devicegroup:
+                break
+            module.fail_json(msg=' \'%s\' device group not found in Panorama. Is the name correct?' % devicegroup)
+    else:
+        device = pandevice.firewall.Firewall(ip_address, username, password, api_key=api_key)
+
+    if security_rule_exists(device, rule_name):
+        module.fail_json(msg='Rule with the same name already exists.')
+
+    try:
+        sec_rule = create_security_rule(
+            rule_name=rule_name,
+            description=description,
+            tag=tag,
+            from_zone=from_zone,
+            to_zone=to_zone,
+            source=source,
+            source_user=source_user,
+            destination=destination,
+            category=category,
+            application=application,
+            service=service,
+            hip_profiles=hip_profiles,
+            group_profile=group_profile,
+            antivirus=antivirus,
+            vulnerability=vulnerability,
+            spyware=spyware,
+            url_filtering=url_filtering,
+            file_blocking=file_blocking,
+            data_filtering=data_filtering,
+            wildfire_analysis=wildfire_analysis,
+            log_start=log_start,
+            log_end=log_end,
+            rule_type=rule_type,
+            action=action
+        )
+
+        changed = add_security_rule(device, sec_rule)
+    except PanXapiError:
+        exc = get_exception()
+        module.fail_json(msg=exc.message)
+
+    if changed and commit:
+        result = _commit(device, devicegroup)
+
+    module.exit_json(changed=changed, msg="okey dokey")
```

Comment:
We could return the result but it is probably too much for average Ansible user. I will stick with our typical error message we used before.